opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instances_v3.go

```diff
@@ -0,0 +1,378 @@
+								Schema: map[string]*schema.Schema{
+									"size": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"type": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"disk_encryption_id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"backup_strategy": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"start_time": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"keep_days": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"nodes": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"availability_zone": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"name": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"role": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"status": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"private_ips": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"public_ips": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"created": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceRdsInstanceV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*cfg.Config)
+	client, err := config.RdsV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmterr.Errorf(errCreateClient, err)
+	}
+
+	listOpts := instances.ListOpts{
+		Name:          d.Get("name").(string),
+		Type:          d.Get("type").(string),
+		DataStoreType: d.Get("datastore_type").(string),
+		VpcId:         d.Get("vpc_id").(string),
+		SubnetId:      d.Get("subnet_id").(string),
+	}
+
+	pages, err := instances.List(client, listOpts).AllPages()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	allInstances, err := instances.ExtractRdsInstances(pages)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var instancesToSet []map[string]interface{}
+	var instancesIds []string
+
+	for _, item := range allInstances.Instances {
```

Comment:
in `data_source/opentelekomcloud_rds_instance_v3` we should return only single instance

```
	if len(allInstances) < 1 {
		return common.DataSourceTooFewDiag
	}

	if len(allInstances) > 1 {
		return common.DataSourceTooManyDiag
	}
```